### PR TITLE
Fix govulncheck CI check on release-3.5

### PR DESCRIPTION
**Pull request by henrybear327**
This commit fixed the Go Vulnerability Checker CI job, which isn't scanning for all go.mod files within the project.

Reference:
- https://github.com/etcd-io/etcd/discussions/18168

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by henrybear327** [comment]
Follow up on https://github.com/etcd-io/etcd/issues/17703

File: .github/workflows/govuln.yaml [hunk 1]
```diff
@@ -16,4 +16,18 @@ jobs:
...
           set -euo pipefail
 
-          go install golang.org/x/vuln/cmd/govulncheck@latest && govulncheck ./...
+          go install golang.org/x/vuln/cmd/govulncheck@latest
+
+          # Find all go.mod files recursively, as we have multiple of them within the project
+          gomod_files=$(find . -name go.mod)
+
+          # Loop through each go.mod file path
+          for filepath in $gomod_files; do
```
Comment (ivanvc) [inline .github/workflows/govuln.yaml:25]: Please refer to https://github.com/etcd-io/etcd/pull/18171#discussion_r1639012673

**Review by ahrtr — APPROVED** [review]
LGTM

Please also update 3.5 changelog

**Review by ivanvc — APPROVED** [review]
LGTM. Thanks, @henrybear327

**Comment by henrybear327** [comment]
> Update release-3.5 changelog #18178

https://github.com/etcd-io/etcd/pull/18178

Done as requested! :)

**Pull request merged**